internal/customtype/customtype.go

```diff
@@ -0,0 +1,35 @@
+package customtype
+
+import (
+	"fmt"
+	"go/ast"
+	"reflect"
+	"sync"
+)
+
+var (
+	customTypesMux sync.Mutex
+	customTypes    = make(map[reflect.Type]func(any) ast.Expr)
+)
+
+// See the top-level valast.RegisterType docstring for details.
+func Register[T any](render func(value T) ast.Expr) {
+	customTypesMux.Lock()
+	var zero T
+	t := reflect.TypeOf(zero)
+	if _, exists := customTypes[t]; exists {
+		panic(fmt.Sprintf("%T already registered", zero))
+	}
+	customTypes[t] = func(value any) ast.Expr { return render(value.(T)) }
+	customTypesMux.Unlock()
+}
+
+// Is indicates if the given reflect.Type has a custom AST representation
+// generator registered.
+func Is(rt reflect.Type) (func(any) ast.Expr, bool) {
+	customTypesMux.Lock()
+	defer customTypesMux.Unlock()
+
+	t, ok := customTypes[rt]
+	return t, ok
+}
```

Comment:
sorry @bobheadxi I think I should've been more clear - my point is I'd rather not have a tiny 35-line one-file package and would prefer this just lives in the root under a `customtype.go` file

does that make sense?

Reply:
I mentioned some reasoning in https://github.com/hexops/valast/pull/27#discussion_r1690500390 - it makes the interface to the rest of the valast codebase clear, and avoids adding globals to the root package internally